File: jelly-runtime/src/store/dapp/access/chain_identity.ts
import { Chain } from './chain';

export interface ChainIdentity {
    chain: Chain; // chain
    identity: string; // Identity address
}

export interface VerifiedChainIdentity {
    chain: Chain; // chain
    identity: string; // Identity address

    // =============== Verification information ===============
    message: string;
    signature: string;
}

File: jelly-model/src/store/dapp/access/chain_identity.rs
use serde::{Deserialize, Serialize};

/// Chain identity
#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
pub struct ChainIdentity {
    /// chain
    pub chain: super::chain::Chain,
    /// Identity address
    pub identity: String,
}

/// Chain authentication
#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
pub struct VerifiedChainIdentity {
    /// chain
    pub chain: super::chain::Chain,
    /// Identity address
    pub identity: String,

    // =============== Verification information ===============
    /// Signature information
    pub message: String,
    /// signature
    pub signature: String,
}
